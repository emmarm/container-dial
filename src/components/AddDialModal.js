import React from 'react';
import PropTypes from 'prop-types';
import Modal from 'react-modal';

const AddDialModal = ({ handleHideAddDialModal, container, isOpen, theme }) => (
  <Modal
    isOpen={isOpen}
    onRequestClose={handleHideAddDialModal}
    closeTimeoutMS={150}
    className="add-dial-modal"
  >
    <h2 className="add-dial-modal__title">
      New{' '}
      <span className="title__span" style={{ color: theme.primary }}>
        {container.name}
      </span>{' '}
      Dial
    </h2>

    <form className="add-dial-modal__form">
      <div className="add-dial-modal__field">
        <label htmlFor="site-name" className="field__label">
          Site Name
        </label>
        <input
          id="site-name"
          className="field__input"
          type="text"
          placeholder="e.g. Facebook"
        />
      </div>

      <div className="add-dial-modal__field">
        <label htmlFor="site-url" className="field__label">
          Site URL
        </label>
        <input
          id="site-url"
          className="field__input"
          type="text"
          placeholder="e.g. https://facebook.com"
        />
      </div>

      <div className="add-dial-modal__buttons">
        <button
          onClick={handleHideAddDialModal}
          className="add-dial-modal__button button--secondary"
        >
          Cancel
        </button>

        <button
          className="add-dial-modal__button button--primary"
          style={{ backgroundColor: theme.primary }}
        >
          Add Dial
        </button>
      </div>
    </form>
  </Modal>
);

AddDialModal.propTypes = {
  handleHideAddDialModal: PropTypes.func.isRequired,
  isOpen: PropTypes.bool.isRequired,
  container: PropTypes.objectOf(PropTypes.string)
};

export default AddDialModal;
