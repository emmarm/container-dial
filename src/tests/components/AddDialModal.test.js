import React from 'react';
import { shallow } from 'enzyme';

import AddDialModal from '../../components/AddDialModal';

describe('AddDialModal component', () => {
  const props = {
    handleHideAddDialModal: jest.fn(),
    container: {
      name: 'Personal',
      color: 'orange'
    },
    isOpen: true,
    theme: {
      primary: '#333',
      light: '#fff',
      dark: '#000'
    }
  };
  const wrapper = shallow(<AddDialModal {...props} />);

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('displays correct title', () => {
    const title = wrapper.find('.add-dial-modal__title').text();
    expect(title).toBe('New Personal Dial');
  });

  it('renders AddDialForm', () => {
    expect(wrapper.find('Connect(AddDialForm)').length).toBe(1);
  });
});
