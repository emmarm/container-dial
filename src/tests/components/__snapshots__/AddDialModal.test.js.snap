// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddDialModal component matches snapshot 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="add-dial-modal"
  closeTimeoutMS={150}
  isOpen={true}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
  style={
    Object {
      "content": Object {
        "background": "white",
      },
      "overlay": Object {
        "background": "linear-gradient(to bottom, #00033, #00033),
            linear-gradient(to bottom, rgba(0,0,0,0.8), rgba(0,0,0,0.8))",
      },
    }
  }
>
  <h2
    className="add-dial-modal__title"
  >
    New
     
    <span
      className="title__span"
      style={
        Object {
          "color": "#333",
        }
      }
    >
      Personal
    </span>
     
    Dial
  </h2>
  <Connect(AddDialForm)
    handleHideAddDialModal={[Function]}
    theme={
      Object {
        "dark": "#000",
        "light": "#fff",
        "primary": "#333",
      }
    }
  />
</Modal>
`;
